File: lib/features/marketplace/booking/domain/entities/booking_status.dart
enum BookingStatus {
  pending, // Student submitted request.
  confirmed, // Tutor accepted request.
  declined, // Tutor declined the request.
  cancelled, // Student/tutor/system cancelled.
  completed, // Lesson Finished.
}

// student clicks Book:
// Status is pending

// The tutor then sees in the Tutor App:

// Siyanda wants a Mathematics lesson
// 15 September · 15:00–16:00
// Online · R180

// [Accept] [Decline]

// If they accept:

// pending → confirmed

// If they decline:

// pending → declined

File: functions/src/bookings/booking_status.ts
// const blockingStatuses = [
//   "payment_required",
//   "confirmed",
// ] as const ;

export const BookingStatus = {
  pending: "pending",
  paymentRequired: "payment_required",
  confirmed: "confirmed",
  declined: "declined",
  cancelled: "cancelled",
  completed: "completed",
} as const;

export type BookingStatus = typeof BookingStatus[keyof typeof BookingStatus];
